File: ros2/src/march_input_device/src/input_device.cpp
/*Authors: Pleuntje Brons and Andrew Hutani, MIX*/

#include "march_input_device/input_device.hpp"
#include <iostream>

IPD::IPD()
    : m_current_mode (ExoMode::BootUp)
{
    std::cout << "IPD succesfully started" << std::endl;
}

ExoMode IPD::getCurrentMode() const
{
    return m_current_mode;
}

std::set<ExoMode> IPD::getAvailableModes() const
{
    return m_available_modes;
}

void IPD::askNewMode() const
{
    std::cout << "Please enter next mode. Available modes are: ";

    for (const auto& mode : m_available_modes) {
        std::cout << mode << ", ";
    }

    std::cout << '\n';
}

void IPD::setCurrentMode(const ExoMode& current_mode)
{
    m_current_mode = current_mode;
}

void IPD::setAvailableModes(const std::set<ExoMode>& available_modes)
{
    m_available_modes = available_modes;
}

File: ros2/src/march_rqt_input_device/march_rqt_input_device/input_device.py
class IPD:
    def __init__(self):
        self._current_mode = None
        self._available_modes = set()

    def get_current_mode(self):
        return self._current_mode

    def get_available_modes(self):
        return self._available_modes

    def ask_new_mode(self):
        # Implement the logic to ask for a new mode here
        pass

    def set_current_mode(self, current_mode):
        self._current_mode = current_mode

    def set_available_modes(self, available_modes):
        self._available_modes = available_modes